std/json.d

```diff
@@ -1710,9 +1701,7 @@ if (isOutputRange!(Out,char))
         }
     }
 
-    toValue = &toValueImpl;
-
-    toValue(root, 0);
```

Comment:
To prevent adding a safety hole in `toJSON`, I suggest adding something like this outside of `toValueImpl`:
```D
// Make the function infer `@system` when `json.put` is `@system`
if (0) json.put(' ');
```

Reply:
I'm not sure how this improves anything.

Reply:
Surely this shouldn't compile:
```D
struct Sink {
    void put(char c) @system
    {
        *(cast(int*) 0xDEADBEEF) = 0;
    }
}

void main() @safe
{
    Sink s;
    auto jv = JSONValue("x");
    toJson(s, jv);
}
```

Reply:
A function that takes a template `OutputRange` cannot be invariably `@trusted`, that's in violation of safe interfaces.

Reply:
All right, but I dislike doing kludges like this. At least it's not as bad as the delegate kludge being replaced.